Rakefile

```diff
@@ -66,25 +66,74 @@ def update_module_github_status!(mod)
   mod[:published]['GitHub'] ||= 'unknown'
   mod[:published]['Puppet Forge'] ||= 'unknown'
 
+  mod_remote = mod[:remote].gsub(/\.git$/,'')
+
   # See if we have a valid release on GitHub
-  github_releases_url = mod[:remote] + '/releases'
+  github_releases_url = mod_remote + '/releases'
 
   begin
     github_query = Nokogiri::HTML(open(github_releases_url).read)
 
     if github_query.xpath('//a/@title').map(&:value).include?(mod[:version])
-      mod[:published]['GitHub'] = 'yes'.green
+      mod[:published]['GitHub'] = 'yes'
     else
-      mod[:published]['GitHub'] = 'no'.red
+      mod[:published]['GitHub'] = 'no'
     end
   rescue
-    mod[:published]['GitHub'] = 'no'.red
+    begin
+      github_releases_url = mod_remote + '/tags'
+      github_query = Nokogiri::HTML(open(github_releases_url).read)
+
+      if github_query.xpath('//a/@title').map(&:value).include?(mod[:version])
+        mod[:published]['GitHub'] = 'tag'
+      else
+        mod[:published]['GitHub'] = 'no'
+      end
+    rescue
+      mod[:published]['GitHub'] = 'no'
+    end
+  end
+
+  if ['yes','tag'].include?(mod[:published]['GitHub'].uncolor)
+    github_diff_uri = mod_remote + "/compare/#{mod[:version]}...HEAD"
+
+    to_keep = [
+      'SIMP/',
+      'build/',
+      'data/',
+      'environments/',
+      'files/',
+      'functions/',
+      'hiera.yaml',
+      'lib/',
+      'manifests/',
+      'metadata.json',
+      'templates/',
+      'types/'
+    ]
+
```

Comment:
For puppet modules I would add:
* tasks
* utils

For assets, I would add:
* GPGKEYS
* bin
* ext
* rsync
* sbin
* share
* scripts
* src